### PR TITLE
[PPP-4283] Use of Vulnerable Components: commons-httpclient-3.0.1.jar…

### PR DESCRIPTION
**Pull request by ppatricio**
… and httpclient-4.0.1.jar (sonatype-2007-0004 | CVE-2011-1498 | CVE-2012-6153 | CVE-2012-5783)

### CODE REVIEW

**Comment by buildguy** [comment]
### :gear: Build finished in 31s with command:
```console
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd \
-Daudit -P !assemblies
```
:exclamation: No tests found!


:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 36s with command:
```console
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd \
-Daudit -P !assemblies
```
:exclamation: No tests found!


:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 31s with command:
```console
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd \
-Daudit -P !assemblies
```
:exclamation: No tests found!


:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 40s with command:
```console
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd \
-Daudit -P !assemblies
```
:exclamation: No tests found!


:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 37s with command:
```console
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd \
-Daudit
```
:exclamation: No tests found!


:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 36s with command:
```console
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd \
-Daudit
```
:exclamation: No tests found!


:information_source:  _This is an automatic message_

**Comment by zhanghaodong8** [comment]
hhh

**Comment by buildguy** [comment]
### :gear: Build finished in 30s with command:
```console
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd \
-Daudit
```
:exclamation: No tests found!


:information_source:  _This is an automatic message_

File: extensions/pom.xml [hunk 1]
```diff
@@ -110,18 +109,6 @@
       <artifactId>commons-fileupload</artifactId>
       <scope>compile</scope>
     </dependency>
-    <dependency>
-      <groupId>org.apache.httpcomponents</groupId>
-      <artifactId>httpclient</artifactId>
-      <version>${httpclient.version}</version>
```
Comment (graimundo) [inline extensions/pom.xml]: Did you double check if the httpclient lib was still being included in the `lib`? It probably isn't it is probably needed...

File: extensions/pom.xml [hunk 2]
```diff
@@ -2122,18 +2109,6 @@
       <version>${annogen.version}</version>
       <scope>runtime</scope>
     </dependency>
-    <dependency>
-      <groupId>org.apache.httpcomponents</groupId>
-      <artifactId>httpcore</artifactId>
-      <version>${httpcore.version}</version>
-      <scope>runtime</scope>
```
Comment (graimundo) [inline extensions/pom.xml]: Did you double check if the httpclient lib was still being included in the `lib`? It probably isn't it is probably needed...

**Comment by buildguy** [comment]
### :x: Build failed in 3m 37s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
Pull request 4518 : Not mergeable at 96cfcb76d62fc4c3cc8a0af7453c15dd5655300d+018eb6e0b0db3bf664268f10d950bb4545357083 (NOT_MERGEABLE)
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 35s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 36s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 37s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 37s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

File: extensions/pom.xml (hunk 1 above)
Comment (ppatricio) [reply extensions/pom.xml]: A pipeline was tested without this, and the lib was still being included.

File: extensions/pom.xml (hunk 2 above)
Comment (ppatricio) [reply extensions/pom.xml]: A pipeline was tested without this, and the lib was still being included.

**Comment by ppatricio** [comment]
We had to include "again" the old commons-httpclient dependency at pentaho.war pom due to axis2-v1.5 dependencies.

**Pull request merged**